docs/general-concepts/logging.md

```diff
@@ -0,0 +1,231 @@
+For example, you can logon to a site where your extension is live, navigate through the site,
+and view your extension's logged messages via a debug console, while other users of the site are unaffected.
+
+There isn't a sample extension to download to demonstrate the functionality,
+but you can easily use one of the [module tutorial](../building-extensions/modules/module-development-tutorial/index.md) steps
+as the basis for experimenting with the functions described below.
+
+Note that Joomla logging is different from Joomla Action Logs (which isn't covered here).
+
+## Basic Logging
+
+To log a message you use the static `add()` function of the Joomla\CMS\Log\Log class. For example: 
+
+```php
+use Joomla\CMS\Log\Log;
+...
+Log::add('my error message', Log::ERROR, 'my-extension-category');
+```
+
+The parameters are
+
+- A message string - whatever you like. 
+You can use language translation with these, for example, `Joomla\CMS\Language\Text::_('MY_EXTENSION_ERR_MSG')` 
+(provided you have loaded your extension's language constants). 
+You can also display the values of variables, provided you convert them into a string format. 
+
+- A priority, which can be one of Log::EMERGENCY, Log::ALERT, Log::CRITICAL, Log::ERROR, Log::WARNING, Log::NOTICE, Log::INFO, Log::DEBUG.
+These are based on the standard syslog / RFC 5424 severity levels – see the [wikipedia article on syslog](https://en.wikipedia.org/wiki/Syslog),
+and defined in libraries/src/Log/Log.php.
+
+- A category, which is just a text string. 
+You can define whatever categories you like, but it is best to define them to avoid possible clashes with those of other extensions.
+Using your extension's name is fine.
+
+:::note
+If you log a message with 'jerror' as the category then the message will also be displayed in the message area on the web page.
+:::
+
+The logging is controller by the administrator functionality, so to see the logged message you must switch logging on.
+
+In the admin Global Configuration / Logging tab:
+
+- ensure that the path to log folder is set (by default it's an absolute pathname pointing to administrator/logs). 
+
+- set the Log Almost Everything switch to Yes. 
+
+After logging the message in your code you should see it in the `everything.php` file. 
+(All categories of messages except deprecation categories get logged in this file).
+
+## Custom Logging
+
+The `everything.php` file contains messages logged by all the software components of the Joomla instance. 
+If just a subset is wanted then an administrator can use the Custom logging section on the Global Configuration / Logging tab.
+
+Joomla will write to a file `custom-logging.php` any messages which:
+
+- match any of the priorities in the Log Priorities field, and
+- match any of a list of categories in the Log Categories field
+
+when the Log Category Mode is set to Include.
+
+(These messages are still written to `everything.php` as well).
```

Comment:
This only happen when both **Log Almost Everything** and **Custom logging**  is enabled.
As I wrote in previous comment. User can choose only one.